### PR TITLE
normalize URL paths

### PR DESCRIPTION
**Pull request by ajgon**

### CODE REVIEW

File: lib/hawk/http.rb [hunk 1]
```diff
@@ -107,7 +107,9 @@ def request(method, path, options)
 
     private
       def build_url path
-        base.merge(path.sub(/^\//, '')).to_s
+        uri = base.merge(path.sub(/^\//, ''))
+        uri.path.squeeze!
+        uri.to_s
```
Comment (tagliala) [inline lib/hawk/http.rb:112]: this is stripping out double slashes?

Can we have a spec covering the issue this is fixing?

**Review by tagliala — CHANGES_REQUESTED** [review]

**Comment by tagliala** [comment]
New Specs are failing because this is removing double `o`s.

We should probably squeeze the slashes

image

File: lib/hawk/http.rb (hunk 1 above)
Comment (ajgon) [reply lib/hawk/http.rb:112]: Yes, this aims to fix https://github.com/ifad/reticulum-client/issues/18 (and probably some others). I will do it properly (and add tests) later, for now I just needed it to debug people.

File: lib/hawk/http.rb [hunk 2]
```diff
@@ -107,7 +107,9 @@ def request(method, path, options)
 
     private
       def build_url path
-        base.merge(path.sub(/^\//, '')).to_s
+        uri = base.merge(path.sub(/^\//, ''))
+        uri.path.squeeze!
```
Comment (tagliala) [inline lib/hawk/http.rb:111]: ?

```suggestion
        uri = base.merge(path.sub(/^\//, '').squeeze('/')).to_s
```

**Comment by tagliala** [comment]
Superseded by #62

**Pull request closed without merging**